### PR TITLE
Check registry file hash before downloading

### PR DESCRIPTION
**Pull request by OliviaLynn**
For #54; checks the hash of the local registry file against the hash of the remote registry file to avoid unnecessary re-downloading of the registry file.

**Note** - the docs are failing because I've renamed the remote `data-registry.txt` to `data_registry.txt` to match our usage in lephare (default behavior is to save this file as `data_registry.txt`). Until we merge lephare-photoz/lephare-data#12, we won't be able to find this new file.

### CODE REVIEW

File: src/lephare/data_retrieval.py [hunk 1]
```diff
@@ -59,21 +59,46 @@ def download_registry_from_github(url="", outfile=""):
...
+        local_registry_hash = pooch.file_hash(outfile, alg="sha256")
+        registry_hash_url = os.path.splitext(url)[0] + "_hash.sha256"
+        try:
+            response = requests.get(registry_hash_url, timeout=60)
+            if response.status_code == 200:
+                remote_registry_hash = response.text.strip()
+                if local_registry_hash == remote_registry_hash:
+                    print(f"Local registry file is up to date: {outfile}")
+                    return
+        except requests.exceptions.RequestException as e:
```
Comment (drewoldag) [inline src/lephare/data_retrieval.py:98]: This logic looks pretty sound to me, and I think I worked though all the edge cases. The unit test coverage might be a little light, but there's also an argument to be made that, once in place, this part of the code will be pretty stable.

**Review by drewoldag — APPROVED** [review]
This looks good to me. I'm glad that you've got unit tests in place, but I'm not fully confident all the logic is exercised (I could be wrong). As mentioned in the comment, it might be worth expanding the unit tests, or possibly just noting which code paths aren't covered in some comments for later implementation.

File: src/lephare/data_retrieval.py (hunk 1 above)
Comment (OliviaLynn) [reply src/lephare/data_retrieval.py:96]: Combined ifs, and renamed "response" variable to hash_response and (below) registry_response for clarity.

File: tests/lephare/test_data_retrieval_registry.py
Comment (drewoldag) [inline tests/lephare/test_data_retrieval_registry.py:1]: Nice, I like the way you've documented the logic and the tests that cover it.

File: src/lephare/data_retrieval.py [hunk 2]
```diff
@@ -53,34 +53,84 @@ def filter_files_by_prefix(file_path, target_prefixes):
...
+    """
+    local_registry_hash = pooch.file_hash(local_registry_file, alg="sha256")
+    remote_hash_url = os.path.splitext(remote_registry_url)[0] + "_hash.sha256"
+
+    remote_hash_response = requests.get(remote_hash_url, timeout=60)
+    remote_hash_response.raise_for_status()  # Raise exceptions for non-200 status codes
+
+    if remote_hash_response.text.strip() == local_registry_hash:
+        print(f"Local registry file is up to date: {local_registry_file}")
+        return True
```
Comment (drewoldag) [inline src/lephare/data_retrieval.py:91]: Minor point, not worth blocking the merge. You could eliminate the two return statements in favor of a single one. But this function is pretty clear, so probably not worth the extra commit.

**Review by drewoldag — APPROVED** [review]
This looks great, very clean code. Nice

File: src/lephare/data_retrieval.py [hunk 3]
```diff
@@ -86,12 +86,7 @@ def _check_registry_is_latest_version(remote_registry_url, local_registry_file):
     remote_hash_response = requests.get(remote_hash_url, timeout=60)
     remote_hash_response.raise_for_status()  # Raise exceptions for non-200 status codes
 
-    if remote_hash_response.text.strip() == local_registry_hash:
-        print(f"Local registry file is up to date: {local_registry_file}")
-        return True
-    else:
-        print(f"Local registry file is not up to date: {local_registry_file}")
-        return False
+    return remote_hash_response.text.strip() == local_registry_hash
```
Comment (drewoldag) [inline src/lephare/data_retrieval.py:89]: Nice
